diff --git a/sites/all/modules/contrib/field_collection_table/field_collection_table.module b/sites/all/modules/contrib/field_collection_table/field_collection_table.module
index fe65d88..9bf9cd9 100644
--- a/sites/all/modules/contrib/field_collection_table/field_collection_table.module
+++ b/sites/all/modules/contrib/field_collection_table/field_collection_table.module
@@ -393,3 +393,12 @@ function field_collection_table_field_widget_settings_form($field, $instance) {
 
   return $form;
 }
+
+/*
+ * Implements hook_form_FORM_ID_alter().
+ */
+function field_collection_table_form_field_ui_field_edit_form_alter(&$form, &$form_state) {
+  if(isset($form['#instance']) && isset($form['#instance']['widget']) && isset($form['#instance']['widget']['type']) && $form['#instance']['widget']['type'] == 'field_collection_table') {
+    unset($form['instance']['default_value_widget']);
+  }
+}
