diff --git a/sites/all/modules/contrib/autosave/autosave.js b/sites/all/modules/contrib/autosave/autosave.js
index 1a1c399..bc8f6be 100644
--- a/sites/all/modules/contrib/autosave/autosave.js
+++ b/sites/all/modules/contrib/autosave/autosave.js
@@ -1,4 +1,5 @@
 (function ($) {
+$(function(){
 
 var showingRestoreCommand;
 
@@ -108,4 +109,5 @@ Drupal.behaviors.autosave.displayMessage = function(message, settings) {
   setTimeout(Drupal.behaviors.autosave.hideMessage, settings.timeout);
 };
 
+});
 })(jQuery);
