diff --git a/handlers/views_handler_filter_combine.inc b/handlers/views_handler_filter_combine.inc
index 915924b..c1afcf5 100644
--- a/handlers/views_handler_filter_combine.inc
+++ b/handlers/views_handler_filter_combine.inc
@@ -54,6 +54,11 @@ class views_handler_filter_combine extends views_handler_filter_string {
     $fields = array();
     // Only add the fields if they have a proper field and table alias.
     foreach ($this->options['fields'] as $id) {
+      // Field access checks may have removed this handler.
+      if (!isset($this->view->field[$id])) {
+        continue;
+      }
+
       $field = $this->view->field[$id];
       // Always add the table of the selected fields to be sure a table alias
       // exists.
