diff --git a/sites/all/modules/contrib/views/includes/handlers.inc b/sites/all/modules/contrib/views/includes/handlers.inc
index 6c1e887..f808eff 100644
--- a/sites/all/modules/contrib/views/includes/handlers.inc
+++ b/sites/all/modules/contrib/views/includes/handlers.inc
@@ -223,8 +223,8 @@ class views_handler extends views_object {
     $this->localization_keys = array(
       $display_id,
       $this->handler_type,
-      $options['table'],
-      $options['id']
+      !empty($options['table']) ? $options['table'] : NULL,
+      !empty($options['id']) ? $options['id'] : NULL
     );
 
     $this->unpack_options($this->options, $options);
diff --git a/sites/all/modules/contrib/views/plugins/views_plugin_display.inc b/sites/all/modules/contrib/views/plugins/views_plugin_display.inc
index a4b732e..c8c8227 100644
--- a/sites/all/modules/contrib/views/plugins/views_plugin_display.inc
+++ b/sites/all/modules/contrib/views/plugins/views_plugin_display.inc
@@ -932,7 +932,7 @@ class views_plugin_display extends views_plugin {
           $info = $this->view->temporary_options[$type][$id];
         }
 
-        if ($info['id'] != $id) {
+        if (empty($info['table']) || $info['id'] != $id) {
           $info['id'] = $id;
         }
 
@@ -957,7 +957,7 @@ class views_plugin_display extends views_plugin {
           $handler_type = $type;
         }
 
-        $handler = views_get_handler($info['table'], $info['field'], $handler_type, $override);
+        $handler = views_get_handler(!empty($info['table']) ? $info['table'] : NULL, !empty($info['field']) ? $info['field'] : NULL, $handler_type, $override);
         if ($handler) {
           // Special override for area types so they know where they come from.
           if ($handler_type == 'area') {
@@ -2914,7 +2914,7 @@ class views_plugin_display extends views_plugin {
           $override = $aggregate[$info['group_type']]['handler'][$type];
         }
       }
-      $handler = views_get_handler($info['table'], $info['field'], $handler_type, $override);
+      $handler = views_get_handler(!empty($info['table']) ? $info['table'] : NULL, !empty($info['field']) ? $info['field'] : NULL, $handler_type, $override);
       if ($handler) {
         $handler->init($this->view, $info);
         $output .= $indent . '/* ' . $types[$type]['stitle'] . ': ' . $handler->ui_name() . " */\n";
