diff --git a/sites/all/modules/contrib/linkchecker/linkchecker.pages.inc b/sites/all/modules/contrib/linkchecker/linkchecker.pages.inc
index 7bc16f8..39f8ca9 100644
--- a/sites/all/modules/contrib/linkchecker/linkchecker.pages.inc
+++ b/sites/all/modules/contrib/linkchecker/linkchecker.pages.inc
@@ -195,7 +195,7 @@ function _linkchecker_report_page($query, $account = NULL) {
     // Create table data for output.
     $rows[] = array(
       'data' => array(
-        l(_filter_url_trim($link->url, 40), $link->url),
+        l($link->url, $link->url),
         $link->code,
         check_plain($link->error),
         theme('item_list', array('items' => $links)),
